File: logging/src/main/java/com/agmtopy/kocketmq/logging/inner/SerializeType.java
package com.agmtopy.kocketmq.logging.inner;

import com.agmtopy.kocketmq.logging.InternalLogger;

public enum SerializeType {
    JSON((byte) 0),
    ROCKETMQ((byte) 1);

    private byte code;

    SerializeType(byte code) {
        this.code = code;
    }

    public static SerializeType valueOf(byte code) {
        for (SerializeType serializeType : SerializeType.values()) {
            if (serializeType.getCode() == code) {
                return serializeType;
            }
        }
        return null;
    }

    public byte getCode() {
        return code;
    }
}

File: remoting/src/main/kotlin/com/agmtopy/kocketmq/remoting/protocol/SerializeType.kt
package com.agmtopy.kocketmq.remoting.protocol

enum class SerializeType(val code: Byte) {
    JSON(0.toByte()), ROCKETMQ(1.toByte());

    companion object {
        fun valueOf(code: Byte): SerializeType? {
            for (serializeType in values()) {
                if (serializeType.code == code) {
                    return serializeType
                }
            }
            return null
        }
    }
}